velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,321 @@
+    0: {0, 0, 0}
+    1: {1, 5, 6}
+    2: {2, 10, 12}
+    3: {3, 15, 18}
+    4: {4, 20, 24}
+    5: {5, 25, 30}
+
+Let's try a slightly more complex expression: `2 * a + b % 3`.
+
+.. code-block:: c++
+
+    exprSet = compileExpression("2 * a + b % 3", asRowType(data->type()));
+    std::cout << exprSet->toString() << std::endl;
+
+.. code-block::
+
+    > plus(multiply(2:BIGINT, a), mod(b, 3:BIGINT))
+
+.. code-block:: c++
+
+    auto d = evaluate(*exprSet, data);
+
+    auto abcd = makeRowVector({"a", "b", "c", "d"}, {a, b, c, d});
+    std::cout << abcd->toString(0, d->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT,c:BIGINT,d:BIGINT>: 6 elements, no nulls]
+    0: {0, 0, 0, 0}
+    1: {1, 5, 6, 4}
+    2: {2, 10, 12, 5}
+    3: {3, 15, 18, 6}
+    4: {4, 20, 24, 10}
+    5: {5, 25, 30, 11}
+
+You can construct and evaluate arbitrary SQL expressions using :doc:`Presto
+functions <functions>` and special forms like IF, AND, OR, COALESCE, TRY and more.
+If a function you need is not available, feel free to add a new one by following
+the :doc:`How to add a scalar function? <develop/scalar-functions>` guide.
+
+Check out :doc:`Expression Evaluation <develop/expression-evaluation>` article
+to learn what happens under the hood or play with `velox/examples/ExpressionEval.cpp
+<https://github.com/facebookincubator/velox/blob/main/velox/examples/ExpressionEval.cpp>`_
+which contains a detailed step-by-step example of constructing and evaluating
+expression trees using non-test library APIs.
+
+Queries
+-------
+
+Aggregations
+~~~~~~~~~~~~
+
+We can calculate sum and average of 'a' and 'b' by creating and
+executing a query plan with an aggregation node:
+
+.. code-block:: c++
+
+    auto plan = PlanBuilder()
+                  .values({data})
+                  .singleAggregation(
+                      {},
+                      {"sum(a) AS sum_a",
+                       "avg(a) AS avg_a",
+                       "sum(b) AS sum_b",
+                       "avg(b) AS avg_b"})
+                  .planNode();
+
+    auto sumAvg = getResults(plan);
+
+And here are the results:
+
+.. code-block:: c++
+
+    std::cout << sumAvg->toString() << std::endl;
+    std::cout << sumAvg->toString(0, sumAvg->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<sum_a:BIGINT,avg_a:DOUBLE,sum_b:BIGINT,avg_b:DOUBLE>: 1 elements, no nulls]
+    0: {15, 2.5, 75, 12.5}
+
+You can use any of the available :doc:`Presto aggregate functions <functions/aggregate>`
+or add a new one by following :doc:`How to add an aggregate function? <develop/aggregate-functions>`
+guide. Check out :doc:`Aggregations <develop/aggregations>` article for a deep dive
+into aggregation-specific optimizations available in Velox.
+
+Sorting
+~~~~~~~
+
+We can sort data using the OrderBy plan node.
+
+.. code-block:: c++
+
+    plan = PlanBuilder().values({data}).orderBy({"a DESC"}, false).planNode();
```

Comment:
Please state this parameter name in a comment before the value false for clarity.